### DISCUSSION TITLE
Is possible to make some users can read the Workflow designs only, but can not change it?

### DISCUSSION THREAD

**Discussion by kelvenray** [elsa-workflows/elsa-core#4294] [Q&A] [answered]
Here is the scenario.

**- User Role Administrator** should be able to:
1. Add new Workflow
2. Edit Workflow
3. Edit Activities inside the Workflows
4. Delete Workflows

**- User Role Readonly Users** should be able to:
1. Read Workflow list
2. Open a Workflow (via Designer) and check the activities inside
3. Open a activity inside a workflow to read the configuration.
4. The users of this role can read only, cannot Add/Update/Delete any workflows/Activities

How can I implement similiar requirement?

**Comment by sfmskywalker** [comment, marked as answer]
The feature you're asking for, which is role-based access control for workflows and activities, is currently not a part of Elsa Workflows. However, you have a couple of options to achieve this:

1. **Submit a Pull Request (PR)**: Elsa Studio is an open-source project. You can contribute to the project by coding this feature yourself and submitting a PR. The community and project maintainers can then review and potentially merge your contribution into the project.

2. **Build your own dashboard**: You can build your own custom dashboard or user interface that interacts with Elsa Workflows. This custom dashboard would handle user authentication and implement the necessary role-based access control logic. It would provide access to the Elsa Workflows' functionality based on user roles and permissions, thus achieving your goal.

Please be advised that both approaches require a good understanding of the Elsa Workflows' codebase.
